core/src/test/scala/ai/lum/odinson/foundations/TestOdinsonParentQuery.scala

```diff
@@ -0,0 +1,86 @@
+package ai.lum.odinson.foundations
+
+import ai.lum.odinson.{ Document => OdinsonDocument }
+import ai.lum.odinson.ExtractorEngine
+import ai.lum.odinson.lucene.OdinResults
+import ai.lum.odinson.lucene.search.OdinsonQuery
+import ai.lum.odinson.utils.TestUtils.OdinsonTest
+
+class TestOdinsonParentQuery extends OdinsonTest {
+
+  // keys in ai.lum.odinson.utils.TestUtils.ExampleDocs.json
+  val docNames: Seq[String] = Seq("tp-briggs", "tp-pies")
+  val docs: Seq[OdinsonDocument] = docNames.map(getDocument)
+
+  val ee: ExtractorEngine = ExtractorEngine.inMemory(defaultConfig, docs)
+
+  def combineQueries(odinsonPattern: String, parentQuery: String): OdinsonQuery =
+    ee.compiler.mkQuery(
+      pattern = odinsonPattern,
+      parentPattern = parentQuery
+    )
+
+  "ExtractorEngine" should "not return results when pattern succeeds and parent query fails" in {
+    // a simple Odinson pattern
+    val pattern = "[lemma=pie]"
+    // a query (using Lucene query syntax) that is executed against the document metadata
+    val parentQuery: String = "character:Major*"
```

Comment:
Is there a reason for adding the `*` to this query?

Reply:
Ah, so this dates back to when that particular metadata field was a `StringField` with the assumption being that the pattern would be applied to some untokenized text.  Now that this metadata field is indexed as a `TokensField`, this pattern should probably be adjusted to `character:Maj*`.  With this test, we want to illustrate how wildcards can be used with a Lucene query against such a field.